### PR TITLE
visibility for AWS err code in addition to message

### PR DESCRIPTION
**Pull request by surajkota**
Description of changes:
Provide aws error code(e.g. Validation exception, InvalidParameterValue) in addition to message in the condition message

Testing:
Manually introduced an aws error from sdkDelete and verified it gets updated in conditions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by brycahta — APPROVED** [review]
LGTM; will be helpful during development 👍

**Comment by brycahta** [comment]
> Description of changes:
> Provide aws error code(e.g. Validation exception, InvalidParameterValue) in addition to message in the condition message
> 
> Testing:
> Manually introduced an aws error from sdkDelete and verified it gets updated in conditions
> 
> By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

super nit: `Testing` section relates to your other PR not this one

File: templates/pkg/resource/sdk.go.tpl [hunk 1]
```diff
@@ -224,7 +224,7 @@ func (rm *resourceManager) updateConditions (
 			errorMessage = err.Error()
 		} else {
 			awsErr, _ := ackerr.AWSError(err)
-			errorMessage = awsErr.Message()
+			errorMessage = awsErr.Code() + ": " + awsErr.Message()
```
Comment (A-Hilaly) [inline templates/pkg/resource/sdk.go.tpl:227]: small nit: you can also do `fmt.Sprintf("%s: %s")`

Comment (jaypipes) [inline templates/pkg/resource/sdk.go.tpl:227]: @surajkota you can just call [`awsErr.Error()`](https://github.com/aws/aws-sdk-go/blob/2ab2a7fda76a11d2df0a0d15e554de2139672d31/aws/awserr/types.go#L65) above, which automatically [produces the code and message](https://github.com/aws/aws-sdk-go/blob/2ab2a7fda76a11d2df0a0d15e554de2139672d31/aws/awserr/types.go#L13).

**Review by jaypipes — CHANGES_REQUESTED** [review]

File: templates/pkg/resource/sdk.go.tpl (hunk 1 above)
Comment (surajkota) [reply templates/pkg/resource/sdk.go.tpl]: will do

Comment (surajkota) [reply templates/pkg/resource/sdk.go.tpl]: done

**Comment by surajkota** [comment]
> super nit: Testing section relates to your other PR not this one

yes, I am using same setup for testing

**Comment by surajkota** [comment]
@jaypipes @A-Hilaly updated

**Review by jaypipes — APPROVED** [review]

**Pull request merged**